--- yacc/y1.c
+++ yacc/y1.c
@@ -339,7 +339,7 @@ writem(int *pp)
 	/* an item calling for a reduction */
 	if ((i = *pp) < 0) {
 		q = chcopy(q, "    (");
-		sprintf(q, q + isize - sarr, "%d)", -i);
+		snprintf(q, q + isize - sarr, "%d)", -i);
 	}
 	return (sarr);
 }
--- yacc/y2.c
+++ yacc/y2.c
@@ -638,7 +638,7 @@ static const char yyident[] USED = \"yacc: %s\"\n", rel);
 				/* action within rule... */
 
 				lrprnt();		/* dump lhs, rhs */
-				sprintf(actname, sizeof actname,
+				snprintf(actname, sizeof actname,
 						"$$%d", nprod);
 				/*
 				 * make it nonterminal
